File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlFunctionExpression.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlFunctionExpression : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.Function;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_FunctionExpression(this);
    }

    public string FunctionName { get; set; } =string.Empty;
    public List<ISqlExpression> Parameters { get; set; } = [];

    public string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write(FunctionName);
        sql.Write("(");
        for (var i = 0; i < Parameters.Count; i++)
        {
            sql.Write(Parameters[i].ToSql());
            if (i < Parameters.Count - 1)
            {
                sql.Write(", ");
            }
        }
        sql.Write(")");
        return sql.ToString();
    }

}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlFunctionExpression.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression } from './ISqlExpression';

export class SqlFunctionExpression implements ISqlExpression {
    SqlType: SqlType = SqlType.Function;
    Span: TextSpan = new TextSpan();
    Name: string = '';
    Parameters: ISqlExpression[] = [];
    
    Accept(visitor: any): void {
        // 簡單實作
    }

    ToSql(): string {
        return `${this.Name}(${this.Parameters.map(p => p.ToSql()).join(', ')})`;
    }
} 